### ISSUE TITLE
Not working?

### ISSUE THREAD

**Issue by becatlibra** [MA3STR0/kimsufi-crawler#20]
It seems to be giving false positives.  I adding some debugging output into the code, and it doesn't seem to be working at all actually.  I'm a bit confused by this.

Is there any way you can confirm for me that this project is supposed to be still working and if so, can you help me to understand what is going on, on my end?

```
---------------------------------------------------
2015-02-20 06:42:27,397 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,397 SERVER: KS-1 :: ZONE: bhs
2015-02-20 06:42:27,398 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,398 SERVER: KS-1 :: ZONE: rbx
2015-02-20 06:42:27,398 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,398 SERVER: KS-1 :: ZONE: sbg
2015-02-20 06:42:27,398 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,398 SERVER: KS-3 :: ZONE: bhs
2015-02-20 06:42:27,398 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,398 SERVER: KS-3 :: ZONE: rbx
2015-02-20 06:42:27,398 State change - KS-3_available_in_rbx: True
2015-02-20 06:42:27,776 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,776 SERVER: KS-3 :: ZONE: sbg
2015-02-20 06:42:27,776 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,776 SERVER: KS-3 :: ZONE: bhs
2015-02-20 06:42:27,776 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,776 SERVER: KS-3 :: ZONE: rbx
2015-02-20 06:42:27,776 State change - KS-3_available_in_rbx: False
2015-02-20 06:42:27,776 E :: {u'__class': u'dedicatedType:dedicatedAvailability2ZoneStruct', u'zone': u'bhs', u'availability': u'unknown'}
2015-02-20 06:42:27,776 SERVER: KS-3 :: ZONE: sbg
```
Thanks

B

**Comment by MA3STR0** [comment]
I briefly looked at your logs, but sorry- could not identify the issue. I can tell however that the project is alive, because I get notifications sometimes, follow link and check it. I will take a closer look on next week, when I'm back from vacation.

**Comment by becatlibra** [comment]
For reference, the following is the block of code that I changed to get more verbose output.  I added two logger lines.  If I am understanding the JSON correctly, all the lines are referring to zone bhs

```
            # iterate over all tacked zones and set availability states
            for zone in config['zones']:
                _logger.info("E :: %s", e)
                server = SERVER_TYPES[item['reference']]
                _logger.info("SERVER: %s :: ZONE: %s", server, zone)
                state_id = '%s_available_in_%s' % (server, zone)
```

**Comment by MA3STR0** [comment]
I debugged the script carefully and here is what I found:
- script runs correctly
- your example is logging `e` - the variable from list comprehension that is always pointing to last element of `availability` list. That's not relevant to loop you are debugging.
- false-positives may be caused by duplicate entries in OVH api, as one entry sometimes has True availability immediately followed by False. Now script is catching this situation and logs it; if that gets too annoying we can put `return` after _logger.info to retry the whole thing.
